flake.nix

```diff
@@ -23,23 +23,15 @@
     flake-parts,
     git-hooks,
     ...
-  }: let
-    overlay = import ./nix/overlay.nix {inherit self;};
-  in
+  }:
     flake-parts.lib.mkFlake {inherit inputs;} {
-      systems = [
-        "x86_64-linux"
-        "aarch64-linux"
-        "x86_64-darwin"
-        "aarch64-darwin"
-      ];
-      perSystem = {system, ...}: let
-        pkgs = import nixpkgs {
-          inherit system;
-          overlays = [
-            overlay
-          ];
-        };
+      systems = builtins.attrNames nixpkgs.legacyPackages;
+      perSystem = {
+        system,
+        pkgs,
+        ...
+      }: let
```

Comment:
alternatively if you dont like the `pkgs' = pkgs` (which is fine but increases the diff), 
`perSystem = { system, ... }@attrs: let pkgs= attrs.pkgs;`